src/Data/Vault/ST/backends/GHC.hs

```diff
@@ -18,6 +18,10 @@ fromAny = unsafeCoerce
 newtype Vault s = Vault (Map Unique Any)
 newtype Key s a = Key Unique
 
+#if __GLASGOW_HASKELL__ >= 708
+type role Key phantom nominal
```

Comment:
I'm not completely sure about the status of the parameter `s` in the `ST` monad, but I think it should be `nominal` as well. Otherwise, it would be possible to write bad programs using `coerce` as well. One might actually want to report this issue to GHC HQ.